versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5967 @@
+
+```wdl
+task echo {
+  command <<< echo "hello world" >>>
+  output {
+    File message = stdout()
+  }
+}
+```
+
+## File stderr()
+
+Returns the value of the executed command's standard error (stderr) as a `File`. The file should be given a random name and written in a temporary directory, so as not to conflict with any other task output files.
+
+**Parameters**: None
+
+**Returns**: A `File` reference to the stderr generated by the command of the task where the function is called.
+
+**Restrictions**: Can only be used within the `output` section of a `task`.
+
+**Example**
+
+This task echos a message to standard error, and then returns a `File` containing that message.
+
+```wdl
+task echo {
+  command <<< >&2 echo "hello world" >>>
+  output {
+    File message = stderr()
+  }
+}
+```
+
+## Array[File] glob(String)
+
+Returns the Bash expansion of the [glob string](https://en.wikipedia.org/wiki/Glob_(programming)) relative to the task's execution directory, and in the same order.
+
+`glob` finds all of the files (but not the directories) in the same order as would be matched by running `echo <glob>` in Bash from the task's execution directory.
+
+At least in standard Bash, glob expressions are not evaluated recursively, i.e. files in nested directories are not included. 
+
+**Parameters**:
+
+1. `String`: The glob string.
+
+**Returns**: A array of all files matched by the glob.
+
+**Restrictions**: Can only be used within a `task`.
+
+**Example**
+
+```wdl
+task gen_files {
+  command <<<
+    ./generate_files --num 4 --prefix "a"
+  >>>
+
+  output {
+    Array[File] a_files = glob("a*")
+  }
+}
+```
+
+This command generates the following directory structure:
+
+```txt
+execution_directory
+├── a1.txt
+├── ab.txt
+├── a_dir
+│   ├── a_inner.txt
+├── az.txt
+```
+
+Running `echo a*` in the execution directory would expand to `a1.txt`, `ab.txt`, `a_dir` and `az.txt`, in that order. Since `glob` ignores directories, `a_dir` is discarded and the result of the expression is `["a1.txt", "ab.txt", "az.txt"]`.
+
+### Non-standard Bash
+
+The runtime container may use a non-standard Bash shell that supports more complex glob strings, such as allowing expansions that include `a_inner.txt` in the example above. To ensure that a WDL is portable when using `glob`, a container image should be provided and the WDL author should remember that `glob` results depend on coordination with the Bash implementation provided in that container.
+
+## String basename(String|File, [String])
+
+Returns the "basename" of a file - the name after the last directory separator in the file's path. The optional second parameter may specify a suffix to remove from the file name.
+
+**Parameters**
+
+1. (`String`|`File`): Path of the file to read. If the argument is a `String`, it is assumed to be a local file path relative to the current working directory of the task.
+2. `[String]`: Suffix to remove from the file name.
```

Comment:
In cromwell the second string has as a regex possibility. This is useful for certain types of files. For example: `\.ya?ml` matching `.yml` and `.yaml`. 

Should that be codified in the spec as well?

Reply:
I think we can consider this. Is the regexp a full POSIX ERE or a simple glob expression?

The larger question is whether we're willing to add new functionality in this version of the spec or if it should go through the PR process. The decision should also apply to the new `unzip` function.

Reply:
If I write `basename(path, ".yml")` does the `.` match any character? (seems counterintuitive for this example of the common usage)

Reply:
Good question/point. If a regexp is allowed, then the '.' always has to be escaped. The alternative would be to introduce a different syntax for regular expressions, e.g. `/regexp/` that can only be used in specific situations.

Reply:
For the people who need `\.ya?ml` there is already `sub(string, string, string)`. So I am inclined to vote for `basename(path, extension)` being kept simple. Even if that breaks some of the files in biowdl/tasks (what do you think @DavyCats ?) . Maybe: basename(path, extension, regex=false) is better. Because regex=true can then be explicitly set. But that would be for a later version.

Reply:
I agree. Let's leave the regexp option out in 1.1 and discuss whether to add a specific regexp syntax in 2.0 or support the third `basename` parameter.